modules/src/ics03_connection/handler/verify.rs

```diff
@@ -154,28 +154,27 @@ pub fn verify_consensus_proof(
             proof.height(),
             &expected_consensus,
         )
-        .map_err(|_| Kind::ConsensusStateVerificationFailure.context(proof.height().to_string()))?)
+        .map_err(|e| {
+            Kind::ConsensusStateVerificationFailure(proof.height()).context(e.to_string())
+        })?)
 }
 
+/// Checks that `claimed_height` is within normal bounds, i.e., fresh enough to fall within the
+/// trusting period, but not newer than the current (actual) height of the local chain.
 pub fn check_client_consensus_height(
     ctx: &dyn ConnectionReader,
     claimed_height: Height,
 ) -> Result<(), Error> {
-    // Fail if the consensus height is too advanced.
     if claimed_height > ctx.chain_current_height() {
-        return Err(Kind::InvalidConsensusHeight
-            .context(claimed_height.to_string())
-            .into());
-    }
-
-    // Fail if the consensus height is too old (outside of trusting period).
-    if claimed_height.value()
+        // Fail if the consensus height is too advanced.
+        Err(Kind::InvalidConsensusHeight(claimed_height).into())
+    } else if claimed_height.value()
         < (ctx.chain_current_height().value() - ctx.chain_consensus_states_history_size() as u64)
     {
-        return Err(Kind::StaleConsensusHeight
-            .context(claimed_height.to_string())
-            .into());
+        // Fail if the consensus height is too old (outside of trusting period).
+        Err(Kind::StaleConsensusHeight(claimed_height).into())
+    } else {
+        // Height check is within normal bounds, check passes.
+        Ok(())
```

Comment:
Curious on the Rust idiomatic way for this. In other languages this is frowned upon:
```
if <expr1> {
    return <err1>
} else if <expr2> {
    return <err2>
...
} else  {
    return <err99>
}
```
And instead:
```
if <expr1> {
    return <err1>
}
if <expr2> {
    return <err2>
}
...
return <err99>
```
I also prefer the latter.

Reply:
I would say that in Rust the former is more idiomatic since in Rust if statements are expressions of their own, but both are fine by me.

Reply:
https://doc.rust-lang.org/rust-by-example/flow_control/if_else.html but I have no preference.

Reply:
But if this were happening in the middle of the function (for eg. an early return with explicit `return` statements, then yeah I would also prefer the latter.

Reply:
> https://doc.rust-lang.org/rust-by-example/flow_control/if_else.html but I have no preference.

yes but there is no return in the if arms

Reply:
I am asking because it is very typical that we start a function body with a number of checks where we return errors if they fail. If we have a lot of checks we end up with a long if-then-elseif-... and I find that harder to read.

Reply:
More importantly though I would like to have consistency in our code base. So I will follow whatever we decide :)

Reply:
Yeah for such checks I agree that the version with `return` is better.